packages/vite/src/node/plugins/vendorAnalysis.ts

```diff
@@ -0,0 +1,244 @@
+        }
+        return (await parseImps(info, trace))[0]
+      }
+      const findExports = async (info: ModuleInfo, trace: string[] = []) => {
+        if (!info.code) return []
+        if (exportsCache.has(info.id)) {
+          return exportsCache.get(info.id)!
+        }
+        return (await parseImps(info, trace))[1]
+      }
+
+      /* find all redirects */
+      const redirects = new Map<string, string>()
+      let modulesAnalyzed = 0
+      const collectRedirectsStart = performance.now()
+      await Promise.all(
+        Array.from(this.getModuleIds()).map(async (module) => {
+          modulesAnalyzed++
+          const info = this.getModuleInfo(module)
+          if (info && info.code) {
+            if (info.isEntry) entrieIds.add(module)
+            // get all imports from module
+            const imports = await findImports(info)
+            for (const imp of imports) {
+              // don't process dynamic imports
+              if (imp.d !== -1 || !imp.n) continue
+              // resolve the absolute id
+              const id = (await this.resolve(imp.n, module))?.id
+              if (!id) continue
+              const statment: string = info.code.substring(imp.ss, imp.se)
+              if (statment.startsWith('export * as')) {
+                // export * as foo from "foo.js"
+                const name = statment.substring(12).split(' ', 2)[0]
+                redirects.set(info.id + '/+/' + name, id + '/+/*')
+              } else if (statment.startsWith('export *')) {
+                // export * from "foo.js"
+                const impInfo = this.getModuleInfo(id)
+                if (!impInfo) continue
+                const exports = await findExports(impInfo)
+                for (const exp of exports) {
+                  redirects.set(info.id + '/+/' + exp, id + '/+/' + exp)
+                }
+              } else {
+                // match export from statement
+                const [, impDef, imps] =
+                  statment.match(/export\s+(?:([^,]+)\s*,\s*)?{([^}]+)/) || []
+                // default import
+                if (impDef) {
+                  redirects.set(info.id + '/+/' + impDef, id + '/+/default')
+                }
+                // named imports
+                if (imps) {
+                  for (const name of imps.split(',')) {
+                    const [, impName, alias] =
+                      name.trim().match(/([\S]+)(?:\s+as\s+([\S]+))?/) || []
+                    if (!impName) continue
+                    const varName = alias || impName
+
+                    redirects.set(
+                      info.id + '/+/' + varName,
+                      id + '/+/' + impName
+                    )
+                  }
+                }
+              }
+            }
+          }
+        })
+      )
+      isDebug && debug(`${timeFrom(collectRedirectsStart)} collect redirects`)
+
+      /* trace all files imported by entry */
```

Comment:
should use one type of commenting for consistency
```suggestion
      // trace all files imported by entry
```